balancer-js/src/modules/pools/pool-types/concerns/weighted/exit.concern.ts

```diff
@@ -0,0 +1,275 @@
+   * @param {string}  singleTokenMaxOut - Optional: token address that if provided will exit to given token
+   * @returns         transaction request ready to send with signer.sendTransaction
+   */
+  buildExitExactBPTIn = ({
+    exiter,
+    pool,
+    bptIn,
+    slippage,
+    shouldUnwrapNativeAsset,
+    wrappedNativeAsset,
+    singleTokenMaxOut,
+  }: ExitExactBPTInParameters): ExitPoolAttributes => {
+    if (!bptIn.length || parseFixed(bptIn, 18).isNegative()) {
+      throw new BalancerError(BalancerErrorCode.INPUT_OUT_OF_BOUNDS);
+    }
+    if (
+      singleTokenMaxOut &&
+      singleTokenMaxOut !== AddressZero &&
+      !pool.tokens.map((t) => t.address).some((a) => a === singleTokenMaxOut)
+    ) {
+      throw new BalancerError(BalancerErrorCode.TOKEN_MISMATCH);
+    }
+    if (!shouldUnwrapNativeAsset && singleTokenMaxOut === AddressZero)
+      throw new Error(
+        'shouldUnwrapNativeAsset and singleTokenMaxOut should not have conflicting values'
+      );
+
+    // Parse pool info into EVM amounts in order to match amountsIn scalling
+    const {
+      parsedTokens,
+      parsedBalances,
+      parsedWeights,
+      parsedTotalShares,
+      parsedSwapFee,
+    } = parsePoolInfo(pool);
+
+    // Replace WETH address with ETH - required for exiting with ETH
+    const unwrappedTokens = parsedTokens.map((token) =>
+      token === wrappedNativeAsset ? AddressZero : token
+    );
+
+    // Sort pool info based on tokens addresses
+    const assetHelpers = new AssetHelpers(wrappedNativeAsset);
+    const [sortedTokens, sortedBalances, sortedWeights] =
+      assetHelpers.sortTokens(
+        shouldUnwrapNativeAsset ? unwrappedTokens : parsedTokens,
+        parsedBalances,
+        parsedWeights
+      ) as [string[], string[], string[]];
+
+    let minAmountsOut = Array(sortedTokens.length).fill('0');
+    let userData: string;
+    let value: BigNumber | undefined;
+
+    if (singleTokenMaxOut) {
+      // Exit pool with single token using exact bptIn
+
+      const singleTokenMaxOutIndex = sortedTokens.indexOf(singleTokenMaxOut);
+
+      // Calculate amount out given BPT in
+      const amountOut = SDK.WeightedMath._calcTokenOutGivenExactBptIn(
+        new OldBigNumber(sortedBalances[singleTokenMaxOutIndex]),
+        new OldBigNumber(sortedWeights[singleTokenMaxOutIndex]),
+        new OldBigNumber(bptIn),
+        new OldBigNumber(parsedTotalShares),
+        new OldBigNumber(parsedSwapFee)
+      ).toString();
+
+      // Apply slippage
+      minAmountsOut[singleTokenMaxOutIndex] = subSlippage(
+        BigNumber.from(amountOut),
+        BigNumber.from(slippage)
+      ).toString();
+
+      userData = WeightedPoolEncoder.exitExactBPTInForOneTokenOut(
+        bptIn,
+        singleTokenMaxOutIndex
+      );
+
+      if (shouldUnwrapNativeAsset) {
+        value = BigNumber.from(amountOut);
```

Comment:
The value field in a transaction should represent: amount of ETH to transfer from sender to recipient (in WEI, a denomination of ETH)
So in the exitPool request it should always be 0 as we're only using BPT.

Reply:
Got it! I was under the impression that all ETH values should be described on the value field. I just saw that the frontend code doesn't even use the value field for exit transactions. I'll remove them.